test/nuts/scale/lotsOfClasses.nut.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { performance } from 'node:perf_hooks';
+import * as path from 'path';
+import { TestSession } from '@salesforce/cli-plugins-testkit';
+import * as fs from 'graceful-fs';
+import { MetadataResolver } from '../../../src';
+import { MetadataConverter } from '../../../src';
+import { ComponentSetBuilder } from '../../../src';
+import { recordPerf } from './perfUtils';
+
+const dirCount = 50;
+const classesPerDir = 100;
+const classCount = dirCount * classesPerDir;
+
+const testName = 'lotsOfClasses';
+
+describe(`handles ${classCount.toLocaleString()} classes (${(
+  classCount * 2
+).toLocaleString()} files across ${dirCount.toLocaleString()} folders)`, () => {
+  let session: TestSession;
+
+  before(async () => {
+    session = await TestSession.create({
+      project: {
+        name: 'lotsOfClasses',
+      },
+      authStrategy: 'NONE',
+    });
+    // create some number of files
+    const classdir = path.join(session.project.dir, 'force-app', 'main', 'default', 'classes');
+    for (let d = 0; d < dirCount; d++) {
+      const dirName = path.join(classdir, `dir${d}`);
+      // eslint-disable-next-line no-await-in-loop
+      await fs.promises.mkdir(dirName);
+      for (let c = 0; c < classesPerDir; c++) {
+        const className = `x${d}x${c}`;
+        // eslint-disable-next-line no-await-in-loop
+        await Promise.all([
+          fs.promises.writeFile(
+            path.join(dirName, `${className}.cls`),
+            `public with sharing class ${className} {public ${className}() {}}`
```

Comment:
Would it be interesting to have a class that is more typical in size?  Maybe somewhere between 100-200 lines?